zerver/lib/actions.py

```diff
@@ -2734,7 +2734,8 @@ def bulk_add_subscriptions(streams: Iterable[Stream],
                            users: Iterable[UserProfile],
                            color_map: Optional[Dict[str, str]]=None,
                            from_stream_creation: bool=False,
-                           acting_user: Optional[UserProfile]=None) -> SubT:
+                           acting_user: Optional[UserProfile]=None,
+                           stream_admin_map: Optional[Dict[int, List[Stream]]]=None) -> SubT:
```

Comment:
Wouldn't be better if instead of `List[Stream]` we had `Set[Stream]`? I think it makes sense to have a list of unique streams where that user is an admin, and for that, a set would be more appropriate. Also, making the check `stream in stream_admin_map[user_profile.id]` below would be faster.

Reply:
`Set` doesn't work well with Django model objects, so we shouldn't use that data structure.  A `Dict` mapping a stream_id to the object could be used if we preferred that for some reason, but I'm not sure why we would.